doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp kern_pass_1
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp apply
+
+Congratulations! Now we are able to use our new kernel in the pipeline. E.g.:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp kern_usg
+
+## Networks inference using Inference-API {#gapi_fb_inference}
+
+Two models from OMZ have been used in this sample: the
+<a href="https://github.com/opencv/open_model_zoo/tree/master/models/intel
+/face-detection-adas-0001">face detector</a>
+and the
+<a href="https://github.com/opencv/open_model_zoo/blob/master/models/intel
+/facial-landmarks-35-adas-0002/description/facial-landmarks-35-adas-0002.md">
+facial landmarks detector</a>.
+
+Let's prepare our chosen networks for being processed by G-API's Inference-API.
+Providing a new network support is quite similar to a new kernel implementation.
+There are three main steps:
+
+### 1) Network definition {#gapi_fb_i_1}
+
+To define a network we should use the specific macro G_API_NET:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp net_decl
+
+Similarly to a kernel macro, it takes three arguments to register a new type.
+They are:
+1. Network interface name --- also serves as a name of a new type defined
+   with this macro;
+2. Network signature      --- an `std::function<>`-like signature which defines
+   inputs and outputs;
+3. Network's unique name  --- used to identify it within the system when the
+   type is stripped.
+
+### 2) Network parametrization {#gapi_fb_i_2}
+
+To set parameters of the network we have to generate a `cv::gapi::ie::Params<>`
+object and then give it to the graph by the compile args we've already
+mentioned:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp net_prop
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_old.cpp apply
+
+Every `cv::gapi::ie::Params<>` object is related to the network specified by its
+template argument. We should pass there the network type we have described in
+the previous step.
+Those objects are able to contain paths to structure and weights files and
+a desirable backend which can be one of the backends supported by
+InferenceEngine ("CPU", "GPU", etc.).
+
+### 3) Network inference inside a graph {#gapi_fb_i_3}
+
+New we are able to use the network in our pipeline! To do that, we should call
+`cv::gapi::infer<>()` setting the network type as a template argument to
+specify the network we want to infer:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp net_usg
+
+## Videosource processing via Streaming-API {#gapi_fb_streaming}
+
+There is no interest in using the most of computer vision pipelines to process
+just a single picture -- a big group of them are created to be launched with a
+videosource as an input. Therefore, it is essential for G-API as an
```

Comment:
> a big group of them are created to be launched with a videosource as an input

Say it easier as "usually CV algorithms are running on a video streams rather than individual pictures"